Gordon360/Controllers/ProfilesController.cs

```diff
@@ -476,6 +497,44 @@ public async Task<ActionResult<FacultyStaffProfileViewModel>> UpdateOfficeHours(
             return Ok(result);
         }
 
+        /// <summary>
+        /// Set visibility of some piece of personal data for user.
+        /// </summary>
+        /// <param name="userPrivacy">Faculty Staff Privacy Decisions (see UserPrivacyViewModel)</param>
+        /// <returns></returns>
+        [HttpPut]
+        [Route("user_privacy")]
+        // [StateYourBusiness(operation = Operation.UPDATE, resource = Resource.PROFILE_PRIVACY)]
+        public async Task<ActionResult<UserPrivacyViewModel>> UpdateUserPrivacyAsync(UserPrivacyViewModel userPrivacy)
+        {
+            var authenticatedUserUsername = AuthUtils.GetUsername(User);
+            // The privacy setting of some fields (such as HomeCountry and Country) are controlled by one button,
+            // so they are conbined and sent by one API request, and we need to split them here.
+            string[] subFields = userPrivacy.Field.Split(' ');
+            foreach (var subField in subFields)
+            {
+                userPrivacy.Field = subField;
+                await _profileService.UpdateUserPrivacyAsync(authenticatedUserUsername, userPrivacy);
+            }
```

Comment:
```suggestion
            string[] subFields = userPrivacy.Field.Split(' ');
            foreach (var subField in subFields)
            {
                userPrivacy.Field = subField;
                await _profileService.UpdateUserPrivacyAsync(authenticatedUserUsername, userPrivacy);
            }
```

why are fields split by spaces? Why are they not passed back as an array?

Reply:
This is because some fields are controlled by the same button, such as `Country` and `HomeCountry`. Whenever an user choose the `Visibility` for Home, both `Country` and `HomeCountry` changed. Thus. I pass `Country HomeCountry` as Field in this case and need to split it by space. (Might not be a good practice but not sure if there is an easy way to improve it..